### DISCUSSION TITLE
get cookie or localstorage on first query

### DISCUSSION THREAD

**Discussion by leccyril** [nuxt/nuxt#16140] [Questions] [answered]
I want to detect when user go on url if a token is present in cookie or localstorage, to be able to directly display as connected instead of redirect to login.

What is the best way to do it ? i tried to use setCookie and init server in plugin but compilation error

in plugins/server-init.server.ts

```
export default defineNuxtPlugin(event => {
	console.log("server init");

	setCookie(event,'test', 'test');
});
```

but on browser i have setCookie is undefined !

**Comment by Diizzayy** [comment, marked as answer]
> but on browser i have setCookie is undefined !

@leccyril  `setCookie` is a nitro (server only) utility, hence it can't be used client-side, you should instead leverage the [`useCookie` composable](https://nuxt.com/docs/api/composables/use-cookie#usecookie)

```ts
//  plugins/server-init.server.ts
export default defineNuxtPlugin((nuxtApp) => {
  const myCookie = useCookie('<cookie-name>') // attempt to retrieve cookie
  const myState = useState('<key>', () => myCookie.value) // attempt to initialize state with cookie's value
})
```

> I want to detect when user go on url if a token is present in cookie or localstorage, to be able to directly display as connected instead of redirect to login.

Given the plugin above, you'd then be able to access `useState('<key>')`, to detect if the cookie is present.

You might also want to look into [route middlewares](https://nuxt.com/docs/guide/directory-structure/middleware), it would allow you to check where a user is navigating to, hence allowing you to run custom logic accordingly

**Reply by leccyril** [reply]
Works well, indeed on each first request we detect if cookie is present, we can validate it, and add in useState for example is connected and redirect where we need
thanks
